Plugin/Model/Order/ItemRepositoryPlugin.php

```diff
@@ -0,0 +1,166 @@
+        $this->getDefaultSourceSelectionAlgorithmCode = $getDefaultSourceSelectionAlgorithmCode;
+        $this->sourceSelectionService = $sourceSelectionService;
+    }
+
+    /**
+     * Set additional attributes when requesting a single order item
+     *
+     * @param ItemRepository  $subject
+     * @param OrderItemInterface|Item $result
+     * @return OrderItemInterface|Item
+     */
+    public function afterGet(ItemRepository $subject, OrderItemInterface $result)
+    {
+        return $this->applyExtensionAttributesToOrderItem($result, $result);
+    }
+
+    /**
+     * Set additional attributes when requesting a list of order items
+     *
+     * @param ItemRepository $subject
+     * @param OrderItemSearchResultInterface $result
+     * @return OrderItemSearchResultInterface
+     */
+    public function afterGetList(ItemRepository $subject, OrderItemSearchResultInterface $result)
+    {
+        foreach ($result->getItems() as $item) {
+            $this->applyExtensionAttributesToOrderItem($item, $result->getItems());
+        }
+
+        return $result;
+    }
+
+    /**
+     * Apply extension attributes to order item
+     *
+     * @param OrderItemInterface $orderItem
+     * @param array $allItems
+     * @return OrderItemInterface
+     */
+    private function applyExtensionAttributesToOrderItem(OrderItemInterface $orderItem, array $allItems) : OrderItemInterface
+    {
+        $orderProduct = $orderItem->getProduct();
+
+        // Ensure we do not process any further if there is no product associated and log order item
+        if (!$orderProduct) {
+            $this->logger->debug('Order item is missing associated product', [
+                'order_id' => $orderItem->getOrderId(),
+                'order_item_id' => $orderItem->getId()
+            ]);
+
+            return $orderItem;
+        }
+
+        if (!$extensionAttributes = $orderItem->getExtensionAttributes()) {
+            $extensionAttributes = $this->orderItemExtensionFactory->create();
+        }
+
+        $extensionAttributes->setSourceCode(
+            $this->getCustomAttributeValue($orderItem->getOrder(), $allItems)
+        );
+
+        $orderItem->setExtensionAttributes($extensionAttributes);
+
+        return $orderItem;
+    }
+
+    /**
+     * @param Order $order
+     * @param array $allItems
+     * @return string
+     */
+    private function getCustomAttributeValue(Order $order, array $allItems) : string
+    {
+        $inventoryRequest = $this->getInventoryRequestFromOrder->execute(
+            $order,
+            $this->sourceSelectionResult->getSelectionRequestItems($allItems)
+        );
+
+        $selectionAlgorithmCode = $this->getDefaultSourceSelectionAlgorithmCode->execute();
+        $sourceSelectionResult = $this->sourceSelectionService->execute($inventoryRequest, $selectionAlgorithmCode);
+
+        return $sourceSelectionResult->getSourceSelectionItems()[0]->getSourceCode();
```

Comment:
This needs to be hardened, what if an empty array is returned? Have a look at what `getSourceSelectionItems` returns by default.

Can you also add a TODO comment here questioning what happens if there's more than one source? We currently don't have a use case for it (the source selection required for Seraphine means there should only be one source) but in the future we may need to figure out which source to get the code for